### PR TITLE
en_mm and object_mm OK

### PR DESCRIPTION
**Pull request by fig02**
mm2 also updated to use object file symbols

### CODE REVIEW

File: assets/xml/objects/object_mm.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    <File Name="object_mm" Segment="6">
+        <Skeleton Name="gRunningManSkel" Type="Flex" LimbType="Standard" Offset="0x5E18"/>
+        <Animation Name="gRunningManRunAnim" Offset="0x0718"/>
+        <Animation Name="gRunningManSitStandAnim" Offset="0x6940"/>
+        <Animation Name="gRunningManSitWaitAnim" Offset="0x6C50"/>
+        <Animation Name="gRunningManSprintAnim" Offset="0x0468"/>
+        <Animation Name="gRunningManExcitedAnim" Offset="0x73A0"/>
+        <Animation Name="gRunningManHappyAnim" Offset="0x8060"/>
+        <Texture Name="gRunningManMouthClosedTex" Outname ="running_man_mouth_closed" Format="ci8" Width="32" Height="16" Offset="0xE30"/>
+        <Texture Name="gRunningManMouthOpenTex" Outname ="running_man_mouth_closed" Format="ci8" Width="32" Height="16" Offset="0xC30"/>
```
Comment (Roman971) [inline assets/xml/objects/object_mm.xml:11]:
```suggestion
        <Texture Name="gRunningManMouthClosedTex" Outname="running_man_mouth_closed" Format="ci8" Width="32" Height="16" Offset="0xE30"/>
        <Texture Name="gRunningManMouthOpenTex" Outname="running_man_mouth_closed" Format="ci8" Width="32" Height="16" Offset="0xC30"/>
```

File: src/overlays/actors/ovl_En_Mm/z_en_mm.c [hunk 2]
```diff
@@ -44,39 +71,541 @@ static ColliderCylinderInit D_80AAEAE0 = {
...
+    static Vec3f headOffset = { 200.0f, 800.0f, 0.0f };
+    EnMm* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mm/func_80AAEA10.s")
+    if (limbIndex == 15) {
+        Matrix_MultVec3f(&headOffset, &this->actor.focus.pos);
+        Matrix_Translate(260.0f, 20.0f, 0.0f, MTXMODE_APPLY);
+        Matrix_RotateY(0.0f, 1);
+        Matrix_RotateX(0.0f, 1);
+        Matrix_RotateZ(2.5132742f, 1);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Mm/z_en_mm.c:607]:
```suggestion
        Matrix_RotateY(0.0f, MTXMODE_APPLY);
        Matrix_RotateX(0.0f, MTXMODE_APPLY);
        Matrix_RotateZ(2.5132742f, MTXMODE_APPLY);
```

File: src/overlays/actors/ovl_En_Mm/z_en_mm.c [hunk 3]
```diff
@@ -44,39 +71,541 @@ static ColliderCylinderInit D_80AAEAE0 = {
...
+            if ((sp26 >= 0) && (sp26 <= 0x140) && (sp24 >= 0) && (sp24 <= 0xF0) && (yawDiff <= 17152.0f) &&
+                (this->unk_1E0 != 3) && func_8002F2CC(&this->actor, globalCtx, 100.0f)) {
+                this->actor.textId = EnMm_GetTextId(this, globalCtx);
+            }
+        }
+    }
+}
+
+s32 EnMm_GetPointCount(Path* pathList, s32 pathNum) {
+    return (pathList + pathNum)->count;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Mm/z_en_mm.c:323]: Can this not be matched with a normal array access?

Comment (fig02) [reply src/overlays/actors/ovl_En_Mm/z_en_mm.c:323]: no it cant, and it happens in multiple places in the file, which tells me its real

**Review by Roman971 — APPROVED** [review]

**Review by Thar0 — APPROVED** [review]
Not required but data could be named:

`D_80AAEAE0` -> `sCylinderInit`
`D_80AAEB18` -> `sDamageTable`
`D_80AAEBA8` -> `sPathInfo`
`D_80AAEBE8` -> `sInitChain`

**Comment by fig02** [comment]
oh yea thanks, meant to do that and forgot. done

**Review by Thar0 — APPROVED** [review]

**Pull request merged**